core/src/main/java/fr/sncf/osrd/envelope_sim/allowances/AbstractAllowanceWithRanges.java

```diff
@@ -431,6 +439,12 @@ private Envelope computeEnvelopeWithLeftJunction(Envelope envelopeSection,
     private Envelope intersectLeftRightParts(EnvelopePart leftPart, EnvelopePart rightPart) {
         if (rightPart == null || leftPart == null)
             throw AllowanceConvergenceException.tooMuchTime();
+        if (leftPart.getMaxSpeed() < rightPart.getMinSpeed()) {
+            // The curves don't intersect at all
+            // This sometimes happens when one part is very short compared to the time step
+            // When it happens we have very little margin to add time, so we throw a `tooMuchTime` error
+            throw AllowanceConvergenceException.tooMuchTime();
+        }
```

Comment:
Okay, that's very odd. I would not have expected this to happen

Reply:
If you want to see what's going on here you can put a breakpoint and run the integration tests. It is indeed quite odd.

Reply:
I've moved the discussion to https://github.com/DGEXSolutions/osrd/issues/1535

Reply:
I'm taking care of this, a new commit will arrive shortly